### PR TITLE
feat(hardware-testing): Autofind hw testing devices

### PR DESCRIPTION
**Pull request by ryanthecoder**
# Overview

# Test Plan

# Changelog

# Review requests

# Risk assessment

### CODE REVIEW

File: hardware-testing/hardware_testing/drivers/pressure_fixture.py [hunk 1]
```diff
@@ -98,6 +103,40 @@ def read_all_pressure_channel(self) -> List[float]:
         return pressure
 
 
+def connect_to_fixture(
```
Comment (andySigler) [inline hardware-testing/hardware_testing/drivers/pressure_fixture.py:106]: if something unexpected happened, raise an error, don't pass

File: hardware-testing/hardware_testing/drivers/asair_sensor.py [hunk 2]
```diff
@@ -72,17 +74,31 @@ def get_serial(self) -> str:
         ...
 
 
-def BuildAsairSensor(simulate: bool) -> AsairSensorBase:
+def BuildAsairSensor(simulate: bool, autosearch: bool = True) -> AsairSensorBase:
```
Comment (andySigler) [inline hardware-testing/hardware_testing/drivers/asair_sensor.py:77]: if something unexpected happened, raise an error, don't pass

**Review by andySigler — CHANGES_REQUESTED** [review]

**Review by sfoster1 — CHANGES_REQUESTED** [review]
This doesn't look like the right approach. It looks like what you're trying to do is
for each sensor,
- iterate through comports
- try to talk to that sensor
- if it doesn't respond right, it's not that sensor

But what if some sensor has coincidentally the same feedback? What if there's a different kind of error than "didn't respond right?"

If these sensors are running their own usb interfaces, we should filter on their vid/pid. If they're not, if all they have is identical-looking serial connections and you cannot tell from the host side which is which, I guess this is ok but I'd be much more strongly looking at instead having a guided flow where you start with everything disconnected and connect them one by one or something.

**Comment by ryanthecoder** [comment]
The issue with using VID/PID is that these devices use serial adapters, and the adapters are the VID/PID that show up. The factory/Dumbo do not use consistent adapters either, so they aren't consistent and/or multiple devices show up as the same vid/pid.
There is currently a flow where the operator gets a list of the connected devices on the terminal and just has to remember which one is which and they absolutely hate this and if they select the wrong one it causes the script to fail and they need to restart the process.

We can't have them connect/disconnect because in the gravimetric/photometric setups they cannot reach the USB ports when the enclosure is over the robot.

I went through the datasheets for the sensors to ensure that these discovery messages won't interfere with each other. I tested this code on a bot that had all of them connected and tried connected to them in several orders to ensure that they don't block up or anything. Not Ideal but kind of the only option.

**Review by sfoster1 — APPROVED** [review]
ok, looks good if we raise an error when we don't find anything

**Review by andySigler — APPROVED** [review]

**Pull request merged**